### PR TITLE
Adding more typing in rfcomm.py

### PR DESCRIPTION
**Pull request by zxzxwu**

### CODE REVIEW

File: bumble/rfcomm.py [hunk 1]
```diff
@@ -263,7 +266,7 @@ def __init__(
         self.window_size = window_size
 
     @staticmethod
-    def from_bytes(data: bytes):
+    def from_bytes(data: bytes) -> 'RFCOMM_MCC_PN':
```
Comment (barbibulle) [inline bumble/rfcomm.py:270]: this should use the class object rather than the name as a string.

File: bumble/rfcomm.py [hunk 2]
```diff
@@ -318,7 +323,7 @@ def __init__(self, dlci: int, fc: int, rtc: int, rtr: int, ic: int, dv: int):
         self.dv = dv
 
     @staticmethod
-    def from_bytes(data: bytes):
+    def from_bytes(data: bytes) -> 'RFCOMM_MCC_MSC':
```
Comment (barbibulle) [inline bumble/rfcomm.py:327]: this should use the class object rather than the name as a string.

File: bumble/rfcomm.py [hunk 3]
```diff
@@ -375,8 +380,12 @@ class DLC(EventEmitter):
     sink: Optional[Callable[[bytes], None]]
 
     def __init__(
-        self, multiplexer, dlci: int, max_frame_size: int, initial_tx_credits: int
-    ):
+        self,
+        multiplexer: 'Multiplexer',
```
Comment (barbibulle) [inline bumble/rfcomm.py:376]: this should use the class object rather than the name as a string.

File: bumble/rfcomm.py [hunk 4]
```diff
@@ -846,7 +855,7 @@ class Client:
     multiplexer: Optional[Multiplexer]
     l2cap_channel: Optional[l2cap.Channel]
 
-    def __init__(self, device, connection) -> None:
+    def __init__(self, device: 'Device', connection: 'Connection') -> None:
```
Comment (barbibulle) [inline bumble/rfcomm.py:830]: this should use the class object rather than the name as a string.

File: bumble/rfcomm.py [hunk 5]
```diff
@@ -886,7 +895,7 @@ async def shutdown(self) -> None:
 class Server(EventEmitter):
     acceptors: Dict[int, Callable[[DLC], None]]
 
-    def __init__(self, device) -> None:
+    def __init__(self, device: 'Device') -> None:
```
Comment (barbibulle) [inline bumble/rfcomm.py:870]: this should use the class object rather than the name as a string.

**Review by barbibulle — APPROVED** [review]

File: bumble/rfcomm.py (hunk 1 above)
Comment (zxzxwu) [reply bumble/rfcomm.py:270]: (Also in other packet classes) This seems like a Pylance issue(I am using VS Code). Methods referencing its own class cannot be resolved, but mypy considers it's fine.

Comment (zxzxwu) [reply bumble/rfcomm.py]: Oh I got it on https://stackoverflow.com/questions/33533148

**Review by barbibulle — APPROVED** [review]

**Pull request merged**